ocrd_tesserocr/recognize.py

```diff
@@ -19,62 +24,251 @@ def process(self):
         model = get_languages()[1][-1] # last installed model
         if 'model' in self.parameter:
             model = self.parameter['model']
             if model not in get_languages()[1]:
                 raise Exception("configured model " + model + " is not installed")
         with PyTessBaseAPI(path=TESSDATA_PREFIX, lang=model) as tessapi:
-            log.info("Using model %s in %s for recognition", model, get_languages()[0])
+            log.info("Using model '%s' in %s for recognition at the %s level", model, get_languages()[0], maxlevel)
+            # tessapi.SetVariable("tessedit_single_match", "0")
+            # 
+            # tessedit_load_sublangs
+            # tessedit_preserve_min_wd_len 2
+            # tessedit_prefer_joined_punct 0
+            # tessedit_write_rep_codes 0
+            # tessedit_parallelize 0
+            # tessedit_zero_rejection 0
+            # tessedit_zero_kelvin_rejection 0
+            # tessedit_reject_mode 0
+            # tessedit_use_reject_spaces 1
+            # tessedit_fix_fuzzy_spaces 1
+            # tessedit_char_blacklist
+            # tessedit_char_whitelist
+            # chs_leading_punct ('`"
+            # chs_trailing_punct1 ).,;:?!
+            # chs_trailing_punct2 )'`"
+            # numeric_punctuation .,
+            # unrecognised_char |
+            # ok_repeated_ch_non_alphanum_wds -?*=
+            # conflict_set_I_l_1 Il1[]
+            # preserve_interword_spaces 0
+            # tessedit_enable_dict_correction 0
+            # tessedit_enable_bigram_correction 1
+            # stopper_smallword_size 2
+            # wordrec_max_join_chunks 4
+            # suspect_space_level 100
+            # suspect_short_words 2
+            # language_model_ngram_on 0
+            # language_model_ngram_order 8
+            # language_model_min_compound_length 3
+            # language_model_penalty_non_freq_dict_word 0.1
+            # language_model_penalty_non_dict_word 0.15
+            # language_model_penalty_punc 0.2
+            # language_model_penalty_case 0.1
+            # language_model_penalty_script 0.5
+            # language_model_penalty_chartype 0.3
+            # language_model_penalty_spacing 0.05
+            # textord_max_noise_size 7
+            # enable_noise_removal 1
+            # classify_bln_numeric_mode 0
+            # lstm_use_matrix 1
+            # user_words_file
+            # user_patterns_file
             for (n, input_file) in enumerate(self.input_files):
                 log.info("INPUT FILE %i / %s", n, input_file)
                 pcgts = from_file(self.workspace.download_file(input_file))
                 # TODO use binarized / gray
                 pil_image = self.workspace.resolve_image_as_pil(pcgts.get_Page().imageFilename)
                 tessapi.SetImage(pil_image)
+                metadata = pcgts.get_Metadata() # ensured by from_file()
+                metadata.add_MetadataItem(
```

Comment:
@VolkerHartmann Here's an example how parameter values could be stored in the PAGE instead of METS or externally.